### ISSUE TITLE
Stack Navigator - on Windows, after navigating to a stack and back, header elements cannot be clicked

### ISSUE THREAD

**Issue by AdaData** [react-navigation/react-navigation#10553]
### Current behavior

_I'm currently only seeing this on Windows, we have a MacOS application (via Catalyst so technically an iOS application) that is unaffected and the snack I linked doesn't actually demo the issue since it doesn't allow you to run on Windows._

If you start on the root screen of a stack navigator, navigate to a different screen, then hit the back button (or navigate to the start by any other means), any header buttons on the root screen will become unclickable. Other screens seem unaffected. 

This seems similar to #7052 but since this is Windows only I figured I'd make a separate issue.

[Minimal repro that is also my example code:](https://snack.expo.dev/_dbMNoLpW)

export const Screen1 = ({
  navigation,
  route,
}: StackScreenProps<any, 'Screen1'>) => {
  return <View style={styles.screens}><Button style={styles.navButtons} title="Click me to go to screen 2!" onPress={() => navigation.navigate('Screen2')} /></View>;
}

export const Screen2 = ({
  navigation,
  route,
}: StackScreenProps<any, 'Screen2'>) => {
  return <View style={styles.screens}><Button style={styles.navButtons} title="Click me to go to screen 1!" onPress={() => navigation.navigate('Screen1')} /></View>;

};

const App = () => {
  const headerButton = <Button title={"Do I work?"} onPress={() => alert("I do!") } />;
  return (
    <NavigationContainer>
      <Stack.Navigator screenOptions={{ headerMode: 'screen', headerRight: () => headerButton}}>
        <Stack.Screen name="Screen1" component={Screen1} />
        <Stack.Screen name="Screen2" component={Screen2} />
      </Stack.Navigator>
    </NavigationContainer>
  );
};

export default App;

https://user-images.githubusercontent.com/60498992/165643099-cae0641c-5db1-4802-a93d-0a9fb8f940ee.mov

### Expected behavior

The obvious, but that header buttons work on the root screen

### Reproduction

https://snack.expo.dev/_dbMNoLpW

### Platform

- [ ] Android
- [ ] iOS
- [ ] Web
- [X] Windows
- [ ] MacOS

### Packages

- [ ] @react-navigation/​bottom-tabs
- [ ] @react-navigation/​drawer
- [ ] @react-navigation/​material-bottom-tabs
- [ ] @react-navigation/​material-top-tabs
- [X] @react-navigation/​stack
- [ ] @react-navigation/​native-stack

### Environment

- I've removed the packages that I don't use

| package                                | version |
| -------------------------------------- | ------- |
| @react-navigation/native               |6.0.10
| @react-navigation/bottom-tabs          |6.3.1
| @react-navigation/drawer               |6.4.1
| @react-navigation/stack                |6.2.1
| react-native-safe-area-context         |3.3.2
| react-native-screens                   |3.10.2
| react-native-gesture-handler           |2.2.0
| react-native-reanimated                |1.13.4
| react-native                           |0.67.2
| node                                   |17.6.0
| npm or yarn                            |1.22.17 (yarn)

**Comment by MaxPMagee** [comment]
This issue was fixed in 0.67.14! Hooray!

https://github.com/microsoft/react-native-windows/issues/9977#issuecomment-1192903320

**Comment by cjhines** [comment]
This is still happening as of `0.68.9`.

```
    "@react-native-async-storage/async-storage": "^1.17.5",
    "@react-navigation/bottom-tabs": "^6.3.2",
    "@react-navigation/native": "^6.0.10",
    "@react-navigation/stack": "^6.2.1",
    "react": "17.0.2",
    "react-native": "0.68.2",
    "react-native-gesture-handler": "^2.4.2",
    "react-native-safe-area-context": "^4.2.5",
    "react-native-screens": "^3.13.1",
    "react-native-windows": "0.68.9",
```
